src/components/forms/CloudInitForm.tsx

```diff
@@ -32,8 +36,68 @@ interface Props {
 }
 
 const CloudInitForm: FC<Props> = ({ formik }) => {
-  const getCloudInitRow = (label: string, name: string, value?: string) => {
-    const metadata = getConfigRowMetadata(formik.values, name);
+  const { openPortal, closePortal, isOpen, Portal } = usePortal({
+    programmaticallyOpen: true,
+  });
+  const [modalName, setModalName] = useState<CloudInitKey | null>(null);
+  const [isReadOnlyModal, setIsReadOnlyModal] = useState(false);
+  const [fieldVersions, setFieldVersions] = useState<Record<string, number>>(
+    {},
+  );
+
+  const networkMeta = getConfigRowMetadata(
+    formik.values,
+    "cloud_init_network_config",
+  );
+  const userMeta = getConfigRowMetadata(formik.values, "cloud_init_user_data");
+  const vendorMeta = getConfigRowMetadata(
+    formik.values,
+    "cloud_init_vendor_data",
+  );
+
+  const getCurrentModalMetadata = () => {
+    if (modalName === "cloud_init_network_config") return networkMeta;
+    if (modalName === "cloud_init_user_data") return userMeta;
+    if (modalName === "cloud_init_vendor_data") return vendorMeta;
+    return null;
+  };
+
+  const getModalTitle = (name: CloudInitKey | null, isReadOnly = false) => {
```

Comment:
Maybe mention the instance name and that this is cloud init data?

Also we might want to clarify this is inherited or local in the title -- though it shouldn't be too long either.

Reply:
Looks great!
Maybe better "For instance (cloud)", where (cloud) is the chip. Or "For profile (foo)". I always try to use the chip with the entity type as I wouldn't expect a user to know the icons by heart, so adding a qualifier seems good opportunity for them to learn the icons. Wdyt?